src/rgw/rgw_rados.cc

```diff
@@ -10608,6 +10674,9 @@ int RGWRados::get_obj_iterate_cb(RGWObjectCtx *ctx, RGWObjState *astate,
       if (!len)
 	  return 0;
     }
+    effective_io_ctx.dup(d->io_ctx);
+  }else {
+    effective_io_ctx.dup(d->tail_io_ctx);
```

Comment:
@Jeegn-Chen 

`dup` here is dangerous for you trigger `aio_operate`, `effective_io_ctx` is on the stack, its private `io_ctx_impl` will be released when function return. You can use `effective_io_ctx = d->tail_io_ctx;`, which share `io_ctx_impl` with `d->tail_io_ctx`

Reply:
nice catch. thx